packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -948,72 +948,92 @@ function getNullableCheck(name: string, type: SdkType) {
+    context,
+    getNullableValidType(property.type),
+    false,
+    true
+  );
+  if (serializeFunctionName) {
+    return `${nullOrUndefinedPrefix}${serializeFunctionName}(${propertyFullName})`;
+  } else if (isAzureCoreErrorType(context.program, property.type.__raw)) {
+    return `${nullOrUndefinedPrefix}${propertyFullName}`;
+  } else {
+    return serializeRequestValue(
+      context,
+      property.type,
+      propertyFullName,
+      !property.optional,
+      getEncodeForType(property.type)
+    );
+  }
 }
-export function getRequestModelMapping(
+
+export function getRequestModelProperties(
   context: SdkContext,
   modelPropertyType: SdkModelType & { optional?: boolean },
   propertyPath: string = "body"
-): RequestModelMappingResult {
-  const props: string[] = [];
+): Array<[string, string]> {
+  const props: [string, string][] = [];
   const allParents = getAllAncestors(modelPropertyType);
   const properties: SdkModelPropertyType[] =
     getAllProperties(modelPropertyType, allParents) ?? [];
   if (properties.length <= 0) {
-    return { propertiesStr: [] };
+    return [];
   }
   for (const property of properties) {
     if (property.kind === "property" && isReadOnly(property)) {
       continue;
     }
-    const dot = propertyPath.endsWith("?") ? "." : "";
-    const serializedName = getPropertySerializedName(property);
-    const propertyPathWithDot = `${propertyPath ? `${propertyPath}${dot}` : `${dot}`}`;
-    const nullOrUndefinedPrefix = getPropertySerializationPrefix(
-      context,
-      property,
-      propertyPath
-    );
 
-    const propertyFullName = getPropertyFullName(
-      context,
-      property,
-      propertyPathWithDot
-    );
-    const serializeFunctionName = buildModelSerializer(
-      context,
-      getNullableValidType(property.type),
-      false,
-      true
-    );
-    if (serializeFunctionName) {
-      props.push(
-        `"${serializedName}": ${nullOrUndefinedPrefix}${serializeFunctionName}(${propertyFullName})`
-      );
-    } else if (isAzureCoreErrorType(context.program, property.type.__raw)) {
-      props.push(
-        `"${serializedName}": ${nullOrUndefinedPrefix}${propertyFullName}`
-      );
-    } else {
-      const serializedValue = serializeRequestValue(
-        context,
-        property.type,
-        propertyFullName,
-        !property.optional,
-        getEncodeForType(property.type)
-      );
-      props.push(`"${serializedName}": ${serializedValue}`);
-    }
+    props.push([
+      getPropertySerializedName(property)!,
+      getSerializationExpression(context, property, propertyPath)
+    ]);
   }
 
-  return { propertiesStr: props };
+  return props;
+}
+
+/**
+ *
+ * This function helps translating an HLC request to RLC request,
+ * extracting properties from body and headers and building the RLC response object
+ */
+interface RequestModelMappingResult {
+  propertiesStr: string[];
+  directAssignment?: boolean;
```

Comment:
I didn't see getRequestModelMapping assign anything to directAssignment, is this expected?

Reply:
It doesn't look like this was getting assigned before this PR either. I can update the function to just return `string[]` since it's never being set.

Reply:
Updated